tests/python/test_dataset_conversion.py

```diff
@@ -0,0 +1,610 @@
+    dataset.add_meta_string(
+        "x_coord_longname",
+        longitude_coord_1d[2]["x_coord_longname"]
+    )
+    # Add attribute data to dataset 
+    dataset.add_meta_string("units",data_attributes_1d['units'])
+    dataset.add_meta_string("longname",data_attributes_1d["longname"])
+    dataset.add_meta_string("convention",data_attributes_1d["convention"])
+    # Add dimension data to dataset
+    dataset.add_meta_string("dim_data_x",longitude_coord_1d[0])
+
+def assert_equality_1d(dataset):
+    """Assert that the original data and data put into the dataset are the same 
+    """
+    
+    # Comparing tensor data extracted from dataset after 
+    # add_metadata_for_xarray call to generated 1D data 
+    assert(dataset.get_tensor("1ddata") == data1d).all()
+    assert(dataset.get_meta_strings("units")[0] == data_attributes_1d['units'])
+    assert(dataset.get_meta_strings("longname")[0] == data_attributes_1d["longname"])
+    assert(dataset.get_meta_strings("convention")[0] == data_attributes_1d["convention"])
+    assert(dataset.get_meta_strings("dim_data_x")[0] == longitude_coord_1d[0])
+    # Comparing tensor data and metadata for coordinate data extracted from dataset after 
+    # add_metadata_for_xarray call to generated 1D coordinate data 
+    assert(dataset.get_tensor("x") == longitude_1d).all()
+    assert(dataset.get_meta_strings("x_coord_units")[0] == longitude_coord_1d[2]["x_coord_units"]) 
+    assert(dataset.get_meta_strings("x_coord_longname")[0] == longitude_coord_1d[2]["x_coord_longname"])
+
+def add_2d_data(dataset):
+    """Add 2D tensor data to dataset
+    """
+    
+    dataset.add_tensor("2ddata",data2d) 
+    # Add coordinate data to dataset
+    dataset.add_tensor("x",longitude_2d)
+    dataset.add_meta_string("x_coord_units",longitude_coord_2d[2]["x_coord_units"])    
+    dataset.add_meta_string("x_coord_longname",longitude_coord_2d[2]["x_coord_longname"])
+    dataset.add_tensor("y",latitude_2d)
+    dataset.add_meta_string("y_coord_units",latitude_coord_2d[2]["y_coord_units"])
+    dataset.add_meta_string("y_coord_longname",latitude_coord_2d[2]["y_coord_longname"])
+    # Add attribute data to dataset 
+    dataset.add_meta_string("units",data_attributes_1d["units"])
+    dataset.add_meta_string("longname",data_attributes_1d["longname"])
+    dataset.add_meta_string("convention",data_attributes_1d["convention"])
+    # Add dimension data to dataset 
+    dataset.add_meta_string("dim_data_x",longitude_coord_2d[0])
+    dataset.add_meta_string("dim_data_y",latitude_coord_2d[0])
+
+def assert_equality_2d(dataset):
+    """ Assert that created data and data put into the dataset are the same
+    """
+    
+    # Comparing tensor data and metadata extracted from dataset after 
+    # add_metadata_for_xarray call to generated 1D data 
+    assert(dataset.get_tensor("2ddata") == data2d).all()
+    assert(dataset.get_meta_strings("units")[0] == "m/s")
```

Comment:
This would be less brittle if you were pulling "m/s" from the reference structure rather than hard-coding it here. The same applies to the next 5 lines or so